beefy-primitives/src/commitment.rs

```diff
@@ -95,6 +96,130 @@ impl<TBlockNumber, TPayload, TSignature> SignedCommitment<TBlockNumber, TPayload
 	}
 }
 
+/// Type to be used to denote placement of signatures
+type BitField = Vec<u8>;
+/// Compress 8 bit values into a single u8 Byte
+const CONTAINER_BIT_SIZE: usize = 8;
+
+/// Temporary representation used for encoding efficiency.
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode)]
+struct TemporarySignatures<TCommitment, TSignature> {
```

Comment:
```suggestion
struct SizeOptimizedSignedCommitment<TCommitment, TSignature> {
```
I'd change the name to make a better relation with `SignedCommitment`, but I'm not particularly happy with the proposed name. Opening discussion for some brainstorming. Maybe `PackedSignedCommitment` or `CompressedSignedCommitment`?

Reply:
`CommitmentCodec`?

Reply:
You are right, something understandable, how about just `CompactSignedCommitment`?

Reply:
`CompactSignedCommitment` sounds perfect to me.